### PR TITLE
SOAR-4712 Supported Versions Validator

### PR DESCRIPTION
**Pull request by aholland-r7**
## Description
A validator that checks the plugin.spec.yaml for a supported versions attribute. If it's not available, or is not a list, it will throw an error. 

## Testing
Successfully tested locally and unit tests created and passing

### CODE REVIEW

**Review by jmcadams-r7 — CHANGES_REQUESTED** [review]
Can you add unit tests for this. I thought there was comprehensive suite of tests in this repo already.

**Review by jmcadams-r7 — DISMISSED** [review]

File: icon_validator/rules/plugin_validators/supported_version_validator.py
Comment (pmara-r7) [inline icon_validator/rules/plugin_validators/supported_version_validator.py:1]: Is this used anywhere? Can't find it. Same with requests on line 5 below

**Review by pmara-r7 — COMMENTED** [review]
Ready to approve after the unused imports(?) comment addressed

**Review by pmara-r7 — APPROVED** [review]
Looks good to me. I think this does the job of making sure there are some supported versions...I assume there is some other validator for making sure a supported version is valid (aka "1.0.0" and not something random like "rapid7)

**Review by lmilby-r7 — CHANGES_REQUESTED** [review]
Nice work, just need to update the setup.py and Readme with version update and changelog

**Review by mrinehart-r7 — APPROVED** [review]
Approved pending Luke's comment - however, do not merge this until @mberezin-r7 's changes are into icon-plugin production.

**Review by lmilby-r7 — DISMISSED** [review]

File: unit_test/plugin_examples/supported_version_validator/plugin.spec_bad_empty.yaml
Comment (mberezin-r7) [inline unit_test/plugin_examples/supported_version_validator/plugin.spec_bad_empty.yaml:9]: lol

**Review by mberezin-r7 — DISMISSED** [review]

File: unit_test/plugin_examples/supported_version_validator/plugin.spec_bad_empty.yaml
Comment (pmara-r7) [reply unit_test/plugin_examples/supported_version_validator/plugin.spec_bad_empty.yaml:9]: How did this pass the profanity validator lol

File: unit_test/test_validate_plugin.py
Comment (pmara-r7) [inline unit_test/test_validate_plugin.py:489]: These two defs have the same name, not sure how python was ok with this

File: unit_test/test_validate_plugin.py
Comment (pmara-r7) [inline unit_test/test_validate_plugin.py:493]: In addition, the method names don't mass the .yaml names here

**Review by pmara-r7 — CHANGES_REQUESTED** [review]

File: unit_test/plugin_examples/supported_version_validator/plugin.spec_bad_empty.yaml
Comment (pmara-r7) [inline unit_test/plugin_examples/supported_version_validator/plugin.spec_bad_empty.yaml:9]:
```suggestion
support: blah
```

File: icon_validator/rules/plugin_validators/supported_version_validator.py
Comment (pmara-r7) [inline icon_validator/rules/plugin_validators/supported_version_validator.py:13]: consider making "supported_versions" a variable rather than magic string

File: unit_test/plugin_examples/supported_version_validator/plugin.spec.yaml
Comment (pmara-r7) [inline unit_test/plugin_examples/supported_version_validator/plugin.spec.yaml:9]:
```suggestion
support: blah
```

File: unit_test/plugin_examples/supported_version_validator/plugin.spec_bad_missing_value.yaml
Comment (pmara-r7) [inline unit_test/plugin_examples/supported_version_validator/plugin.spec_bad_missing_value.yaml:9]:
```suggestion
support: blah
```

**Review by pmara-r7 — CHANGES_REQUESTED** [review]

File: icon_validator/rules/plugin_validators/supported_version_validator.py
Comment (mberezin-r7) [reply icon_validator/rules/plugin_validators/supported_version_validator.py]: Done.

**Review by pmara-r7 — APPROVED** [review]

**Review by mrinehart-r7 — CHANGES_REQUESTED** [review]

**Review by mrinehart-r7 — APPROVED** [review]

**Review by jschipp-r7 — APPROVED** [review]

**Pull request merged**
